src/SWP/Bundle/AnalyticsBundle/Repository/RequestMetricRepository.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\AnalyticsBundle\Repository;
+
+use SWP\Bundle\AnalyticsBundle\Model\RequestMetric;
+
+class RequestMetricRepository extends \Doctrine\ORM\EntityRepository
+{
+    /**
+     * @param RequestMetric $requestMetric Persists object and flushes the entity manager
+     */
+    public function persistAndFlush(RequestMetric $requestMetric)
```

Comment:
It doesn't update the data. It persists the data, and flushes the entity
manager. The guy who writes that blog also writes books on Symfony, so he
probably hasn't suggested anything which isn't OK. Injecting the entity is
generally a bad idea, as it can easily lead to circular references in my
experience.

On Mon, Jun 13, 2016 at 12:20 PM, Rafał Muszyński <notifications@github.com>
wrote:

> In src/SWP/Bundle/AnalyticsBundle/Repository/RequestMetricRepository.php
> <https://github.com/superdesk/web-publisher/pull/78#discussion_r66769488>:
>
> > + * For the full copyright and license information, please see the
> > + * AUTHORS and LICENSE files distributed with this source code.
> > + *
> > + * @copyright 2016 Sourcefabric z.ú.
> > + * @license http://www.superdesk.org/license
> > + */
> > +namespace SWP\Bundle\AnalyticsBundle\Repository;
> > +
> > +use SWP\Bundle\AnalyticsBundle\Model\RequestMetric;
> > +
> > +class RequestMetricRepository extends \Doctrine\ORM\EntityRepository
> > +{
> > +    /**
> > +     * @param RequestMetric $requestMetric Persists object and flushes the entity manager
> > +     */
> > +    public function persistAndFlush(RequestMetric $requestMetric)
>
> repository actually can persist, remove data, but it should not *update*
> the data.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/superdesk/web-publisher/pull/78/files/5fbb2cfc7cad6a7bafa68838427244bcec0eb8fa#r66769488>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ARBtFVxsgu6EOnXFxwbCWEhdwjoyB6PMks5qLS7cgaJpZM4Ix7A->
> .
>